IBM/carbon-addons-iot-react
# feat(table): allow any element to be passed into field value

Closes #916 

**Summary**

- Change `TableRow` value PropType to `PropTypes.any` to allow for more detailed and customized fields

**Change List**
- Changed the TableRow `value` Proptype to `any`
- Added number, string, and boolean type checks to search, sort, and filter reducer functions
- Added React node column to stories to show a new field option
- Added warning to basic story that searching, sorting, and filtering will only work with numbers, strings, and booleans
- Added react node columns to tests to ensure all tests still work
- Combined similar Table stories to help reduce the number of stories

**Acceptance Test (how to verify the PR)**

- Check the following story https://deploy-preview-917--carbon-addons-iot-react.netlify.com/?path=/story/watson-iot-table--stateful-example-with-expansion
  - Try searching, sorting, and filtering and see the component still works correctly
